### ISSUE TITLE
message.json hash format proposal

### ISSUE THREAD

**Issue by koriaf** [ausdigital/ausdigital-tap#21]
As discussed before,
we have "hash" field. It's written in spec about default format as SHA256, but other formats are supported. But currently it's impossible to strictly say which format has been used for this concrete message.
Please check out the https://github.com/multiformats/multihash

So, my proposal, to keep it backward-compatible:
* if `hash` field is a multihash then parse it with multihash format
* if not - assume it's HEX representation of SHA256
* strongly recommend to use multihash for new implementations

@monkeypants, @onthebreeze

**Comment by monkeypants** [comment]
I totally agree that we should be using multihash, that's a great suggestion.

Should we also mandate an encoding, e.g. base58?
